scripts/partition_manager/partition_manager.rst

```diff
@@ -0,0 +1,408 @@
+      The Partition Manager always detects unsatisfiable configurations (no false positives), but it might fail on some valid inputs (false negatives).
+
+   See the following examples of valid and invalid configurations:
+
+   .. code-block:: yaml
+      :caption: Span example 1 (fixed order, cannot work)
+
+      mcuboot:
+         placement:
+            before: [spm, app]
+
+      spm:
+         placement:
+            before: [app]
+
+      foo:
+         span: [mcuboot, app] # This will fail, because 'spm' will be placed between mcuboot and app.
+
+      # Order: mcuboot, spm, app
+
+   .. code-block:: yaml
+      :caption: Span example 2 (ambiguous order)
+
+      mcuboot:
+         placement:
+
+      spm:
+         placement:
+            after: [mcuboot]
+
+      app:
+         placement:
+            after: [mcuboot]
+
+      foo:
+         span: [mcuboot, app] # The order of spm and app is ambiguous in this case, but since
+                              # this span exists, Partition Manager will try to increase the
+                              # likelihood that mcuboot and app are placed next to each other.
+
+      # Order 1: mcuboot, spm, app
+      # Order 2: mcuboot, app, spm
+      # The algorithm should coerce order 2 to make foo work.
+
+   .. code-block:: yaml
+      :caption: Span example 3 (ambiguous order, cannot work)
+
+      mcuboot:
+         placement:
+
+      spm:
+         placement:
+            after: [mcuboot]
+
+      app:
+         placement:
+            after: [mcuboot]
+
+      foo:
+         span: [mcuboot, app]
+
+      bar:
+         span: [mcuboot, spm]
+
+      # Order 1: mcuboot, spm, app
+      # Order 2: mcuboot, app, spm
+      # foo requires order 2, while bar requires order 1.
+
+inside: list
+   This property is the inverse of ``span``.
+   ``partition_name`` is added to the ``span`` list of the first existing phony partition in the list.
+   This property can be set for image or placeholder partitions.
+
+   .. code-block:: yaml
+      :caption: Example for the inside property
+
+      mcuboot:
+         inside: [b0]
+
+      b0:
+         span: [] # During processing, this span will contain mcuboot.
+
+share_size: list
+   This properties defines the size of the current partition to be the same as the size of the first existing partition in the list.
+   This property can be set for image or placeholder partitions.
+   It cannot be used by phony partitions.
+   The list can contain any kind of partition.
+
+   Note that if the target partition is the ``app`` or a partition that spans over the ``app``, the size is effectively split between them, because the size of the ``app`` is dynamically decided.
+
+.. _pm_yaml_preprocessing:
+
+Configuration file preprocessing
+================================
+
+Each :file:`pm.yml` file is preprocessed to resolve symbols from Kconfig and DTS.
+
+The following example shows what kind of preprocessing is possible:
```

Comment:
What do you want to show here? How the file can look before preprocessing (resp. what is supported)? Or how the file looks like after preprocessing?

Reply:
It is here to show that it is required to have the `#include <autoconf.h>` expression, and to show how the `size` property can be set through a Kconfig variable.

Reply:
Where does autoconf.h come from?

Reply:
It is a generated file, it is generated during "configure time" (cmake lingo) when the Kconfig process is executed.